pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/sender/PulsarMetricsSender.java

```diff
@@ -0,0 +1,92 @@
+package org.apache.pulsar.broker.stats.sender;
+
+import com.google.common.collect.Sets;
+import io.netty.util.concurrent.DefaultThreadFactory;
+import org.apache.pulsar.broker.PulsarServerException;
+import org.apache.pulsar.broker.PulsarService;
+import org.apache.pulsar.client.admin.PulsarAdminException;
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.common.naming.NamespaceName;
+import org.apache.pulsar.common.naming.TopicName;
+import org.apache.pulsar.common.policies.data.TenantInfo;
+import org.apache.pulsar.common.stats.Metrics;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.bookkeeper.mledger.util.SafeRun.safeRun;
+
+public class PulsarMetricsSender implements MetricsSender {
+    private static final Logger log = LoggerFactory.getLogger(PulsarMetricsSender.class);
+
+    private PulsarService pulsar;
+    private MetricsSenderConfiguration conf;
+    private ScheduledExecutorService metricsSenderExecutor;
+
+    private TopicName topicToSend;
+
+    private Producer<String> producer;
+
+    public PulsarMetricsSender(PulsarService pulsar, MetricsSenderConfiguration conf) {
+        this.pulsar = pulsar;
+        this.conf = conf;
+        this.metricsSenderExecutor = Executors.newSingleThreadScheduledExecutor(new DefaultThreadFactory("pulsar-metrics-sender"));
+
+        this.topicToSend = TopicName.get(
+                "persistent", NamespaceName.get(this.conf.tenant, "brokers"), this.pulsar.getAdvertisedAddress());
+
+        this.prepareTopics();
+
+        try {
+            this.producer = this.pulsar.getClient().newProducer(Schema.STRING)
+                .topic(this.topicToSend.toString())
+                .enableBatching(true)
+                .producerName("metrics-sender-" + this.pulsar.getAdvertisedAddress())
+                .create();
+        } catch (PulsarClientException | PulsarServerException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void prepareTopics() {
+        TenantInfo tenantInfo = new TenantInfo(
+                pulsar.getConfig().getSuperUserRoles(),
+                Sets.newHashSet(pulsar.getConfig().getClusterName()));
+
+        try {
+            this.pulsar.getAdminClient().tenants().createTenant(topicToSend.getTenant(), tenantInfo);
+            this.pulsar.getAdminClient().namespaces().createNamespace(topicToSend.getNamespace());
+            this.pulsar.getAdminClient().topics().createNonPartitionedTopic(topicToSend.toString());
```

Comment:
@jiazhai @sijie what is the better way to manage tenant/ns/topic directly in pulsar code base? I'm currently using PulsarAdmin WDYT.